.flake8

```diff
@@ -1,6 +1,16 @@
 [flake8]
-select = F
-ignore = E,W,C,F401,F841
+# Ignore style and complexity
+# E: style errors
+# W: style warnings
+# C: complexity
+# F401: module imported but unused
+# F403: import *
+# F811: redefinition of unused `name` from line `N`
+# F841: local variable assigned but never used
+# E402: module level import not at top of file
+# I100: Import statements are in the wrong order
+# I101: Imported names are in the wrong order. Should be
+ignore = E, C, W, F401, F403, F811, F841, E402, I100, I101, D400
```

Comment:
Was this change necessary to deal with the black autoformatter? It seems like an unexpectedly long list of exclusions.

In general I think we should be as strict as possible except where it conflicts with the black formatter.

Reply:
Actually maybe this is fine,is it to with removing `select = F`?

Reply:
I don't know, i just copy pasted to mimic jupyterhub repo.

I recall Min being opinionated towards being relaxed unless it can cause problems in the flake8 checks. I'm personally inclined towards not needing to f

Reply:
`ignore = E, C, W, F401, F841` seems to pass flake8

Of the remainder:
```
# F403: import *
```
Best to avoid in new code as it pollutes the namespace
```
# F811: redefinition of unused `name` from line `N`
```
Potential bug in new code, you've assigned a value to a variable and then assigned another value to that variable without using the first value
```
# E402: module level import not at top of file
# I100: Import statements are in the wrong order
# I101: Imported names are in the wrong order. Should be
```
These are presumably taken care of by `id: reorder-python-imports`?

Reply:
https://github.com/consideRatio/oauthenticator/pull/1 cleans up the unused imports

Reply:
Thanks! Makes sense to me!!

Reply:
It can be a lot of work to make an old repo compliant with flake8, I suspect this is one reason for lots of exclusions. In general once the code is compliant it's not too difficult to keep it that way, I've had several bugs that were detected by flake8 before I'd even run my tests.

Arguably for a repo like oauthenticator where we don't run a lot of the code ourselves we should be even stricter as we're often merging solely based on code review, but that's a separate discussion :smile: . Shall we merge this PR?